OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -125,12 +137,14 @@ public void AddSmudge(CPos loc)
 
 		public void TickRender(WorldRenderer wr, Actor self)
 		{
-			var remove = new List<CPos>();
+			var remove = new HashSet<CPos>();
```

Comment:
This doesn't need to be a set, only unique items will be added in the first place.